### PR TITLE
Sd gitfix jzdonationform2

### PR DESCRIPTION
**Pull request by ifsimicoded**
This is @LayaTaal's last two commit from PR #99 applied on top of the latest master to avoid strange diff's from an older version of master when the jz-donationform branch was originally created.

This should fix validation errors on the donation form per @LayaTaal's original PR.

### CODE REVIEW

File: js/tji-donate.js [hunk 1]
```diff
@@ -20,10 +20,22 @@ jQuery(document).ready(function($){
 	});
 
 	// Verify form details upon submit
-	$('.tji-donation-submit').click(function() {
-		event.preventDefault();
+	$('.tji-donation-submit').click(function(e) {
```
Comment (ifsimicoded) [inline js/tji-donate.js]: @LayaTaal so if you want to use built on validation you can remove your manual checking below and use something like 
```
$('#js-donationform').on('submit', function(e){
   e.preventDefault();
  //submission code here
});
```
(I prefixed the id with js for convention)

and for the html
```
<input type="text" name="first_name" id="first_name" required minlength="3" maxlength="25">
<input type="email" name="email" id="email" required>
```
For donation amounts, you'll have to use vanilla js to check for a numeric value > 0?

and for the css
```
.input.required:invalid {
  border:1px solid $tji-red-or-whatever-var-we-have-for-red;
}
```
or similar

File: js/tji-donate.js [hunk 2]
```diff
@@ -20,10 +20,22 @@ jQuery(document).ready(function($){
...
+				$('label[for="' + this.id + '"]').children('.donation-form__error').css("display", "inline-block");
+				error = true;
+				return false;
+			} else {
+				$('label[for="' + this.id + '"]').children('.donation-form__error').css("display", "none");
+			}
+		});
 		
-		if ($('.selected').length) {
+		if (error == false && $('.selected').length) {
```
Comment (ifsimicoded) [inline js/tji-donate.js]: so I think we should check to see that the donation amount is a number and greater than 0 if someone puts a custom amount, right?

**Comment by LayaTaal** [comment]
All, I've updated the code to use HTML validation instead of Javascript for basic input types. I have also added a test to check if a custom donation amount is zero.

Currently, fractional values (i.e. $1.50) are being rounded down to the nearest dollar amount, I'm not sure why this is happening. Can I get someone to look into it?

File: js/tji-donate.js [hunk 3]
```diff
@@ -20,14 +20,21 @@ jQuery(document).ready(function($){
...
+		e.preventDefault();
+
 		if ($('.selected').length) {
 			var payFee = document.getElementById('tax').checked,
 					donation = parseInt($('.selected').val()),
 					fee = 0,
 					total;
+
+			/* Check for valid donation amount */
+			if (isNaN(donation) || donation == 0) {
```
Comment (ifsimicoded) [inline js/tji-donate.js]: let's try `|| donation <= 0`

Comment (LayaTaal) [reply js/tji-donate.js]: Hmm, why !isNan? I used "donation == 0" because the pattern matching in the html actually catches negative numbers. Should I go ahead and test here also?

Comment (ifsimicoded) [reply js/tji-donate.js]: ah, I didn't see the validation below since it's not part of the PR. I typo-ed !isNan, and then updated the comment.
So a couple picky things about the custom donation amount validation message -- 
1) could it be cleared if a user clicks another amount button, or if the user refocuses the amount input box.
2) are we intentionally dropping any decimal contributions? so if someone enters 10.50 we drop the 50 by using parseInt?

File: template-donation.php [hunk 4]
```diff
@@ -178,6 +179,9 @@
 
 ?>
 
+<!-- Including a newer version of jQuery here needed for PayPal functionality -->
+<!-- We can remove the redundant jQuery include once we move away from WordPress -->
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
```
Comment (ifsimicoded) [inline template-donation.php]: Is there any reason we're holding on to the wp older version of jquery. Can we not just remove it from functions.php tji_scripts() and user the updated version you use here?

File: js/tji-donate.js (hunk 3 above)
Comment (LayaTaal) [reply js/tji-donate.js]: Sure, these are good ideas. I will update the code now.

File: template-donation.php (hunk 4 above)
Comment (LayaTaal) [reply template-donation.php]: Reading about that, it looks like it's highly recommended not to change jquery versions due to some core WordPress functions that apparently break. I don't know how serious those issues would be for us, but didn't to go down that path.

Since we plan to move away from WordPress (which will also help address the issue regarding the database), I thought we can do it this way for the time being.

File: js/tji-donate.js (hunk 3 above)
Comment (ifsimicoded) [reply js/tji-donate.js]: Oh, just saw your comment about rounding confusion. This is happening because you're using parseInt on the inbound value. You can use -- Number.parseFloat(val).toFixed(2);

File: js/tji-donate.js [hunk 5]
```diff
@@ -7,9 +7,10 @@ jQuery(document).ready(function($){
 	$('.donation-btn').on('click', function(e) {
 		e.preventDefault();
 		$('.donation-btn').removeClass('selected');
-		$('#other_amount').removeClass('selected');
-		$('.amount-sign').removeClass('amount-sign--focus');
-		$('#other_amount').val('');
+		$('#other_amount')
+			.removeClass('selected')
+			.val('');
```
Comment (ifsimicoded) [inline js/tji-donate.js]: nice
